File: cpp/design-patterns/adapter/naive/checkout_naive.cpp
std::string Checkout::buy(double amountUsd) {
    // ❌ each caller must remember the quirks by hand:
    int cents = (int)(amountUsd * 100 + 0.5);          // dollars -> cents
    auto [status, txn] = bank.makePayment(cents, "USD");
    if (status != 0) throw std::runtime_error("declined");  // 0 == success (?!)
    return txn;
}

bool Refunds::refund(double amountUsd) {
    // ❌ ...and the SAME conversion is copy-pasted here:
    int cents = (int)(amountUsd * 100 + 0.5);
    auto [status, txn] = bank.makePayment(-cents, "USD");
    return status == 0;
}

File: go/design-patterns/adapter/naive/checkout_naive.go
func (c *Checkout) Buy(amountUsd float64) (string, error) {
    // ❌ each caller must remember the quirks by hand:
    cents := int(amountUsd*100 + 0.5)                 // dollars -> cents
    status, txn := c.bank.MakePayment(cents, "USD")
    if status != 0 {                                  // 0 means success (?!)
        return "", errors.New("declined")
    }
    return txn, nil
}

func (r *Refunds) Refund(amountUsd float64) bool {
    // ❌ ...and the SAME conversion is copy-pasted here:
    cents := int(amountUsd*100 + 0.5)
    status, _ := r.bank.MakePayment(-cents, "USD")
    return status == 0
}